File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.util.*;

class Solution {
    public int solution(int[][] routes) {
        Arrays.sort(routes, (r1, r2) -> r1[0] - r2[0]);
        int min = routes[0][1];
        int answer = 1;
        for (int[] route : routes) {
            if (min >= route[0])
                min = Math.min(min, route[1]);
            else {
                min = route[1];
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> routes) {
    sort(routes.begin(), routes.end(), [](const vector<int>& a, const vector<int>& b) {
        return a[1] < b[1];
    });
    int camera = routes[0][1];
    int answer = 1;
    for (auto r : routes) {
        if (r[0] > camera) {
            answer++;
            camera = r[1];
        }
    }
    return answer;
}
